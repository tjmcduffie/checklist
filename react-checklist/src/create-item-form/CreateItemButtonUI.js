import classnames from 'classnames';
import React from 'react';

import '../util/Button.css';

const CreateItemButtonUI = ({className}) => {
  return (
    <button
      className={classnames('grey-button', className)}
      type="submit">
      Add
    </button>
  );
}

export default CreateItemButtonUI;
