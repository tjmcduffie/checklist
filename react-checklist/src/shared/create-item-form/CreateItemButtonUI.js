import classnames from 'classnames';
import React from 'react';

import '../base/css/Button.css';

const CreateItemButtonUI = ({className}) => {
  return (
    <button
      className={classnames('grey-button', className)}
      type="submit">
      Add
    </button>
  );
}

export default CreateItemButtonUI;
