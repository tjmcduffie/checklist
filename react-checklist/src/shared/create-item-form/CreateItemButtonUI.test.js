import CreateItemButtonUI from './CreateItemButtonUI';
import React from 'react';
import ReactDOM from 'react-dom';
import {create} from 'react-test-renderer';

describe('CreateItemButtonUI', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <CreateItemButtonUI className="test" />,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  describe('renders predictably', () => {
    it('when no className is provided', () => {
      const elem = create(
        <CreateItemButtonUI />
      );
      const tree = elem.toTree();
      expect(tree.rendered.props.className).toBe('grey-button');
      expect(elem.toJSON()).toMatchSnapshot();
    });

    it('when a className is provided', () => {
      const elem = create(
        <CreateItemButtonUI className="test" />
      );
      const tree = elem.toTree();
      expect(tree.rendered.props.className).toBe('grey-button test');
      expect(elem.toJSON()).toMatchSnapshot();
    });
  });
});
