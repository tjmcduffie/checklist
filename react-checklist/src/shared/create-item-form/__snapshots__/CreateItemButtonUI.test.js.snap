// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateItemButtonUI renders predictably when a className is provided 1`] = `
<button
  className="grey-button test"
  type="submit"
>
  Add
</button>
`;

exports[`CreateItemButtonUI renders predictably when no className is provided 1`] = `
<button
  className="grey-button"
  type="submit"
>
  Add
</button>
`;
